### PR TITLE
Quote workItemName and consoleUri

### PR DESCRIPTION
**Pull request by lukas-lansky**
Seen in dotnet/performance:

```
[...]: error : Work item SDK ASP.NET MVC App Template netcoreapp3.1 Clean Build in job 2db02a82-cc0b-4316-a9ee-e1489413e4a4 has failed. [...]
[...]: error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/2db02a82-cc0b-4316-a9ee-e1489413e4a4/workitems/SDK ASP.NET MVC App Template netcoreapp3.1 Clean Build/console [...]
```

It seems to me that the logs would get more readable if it were clear where the work item and URL ends.

### CODE REVIEW

**Review by alexperovich — CHANGES_REQUESTED** [review]
Quoting this won't make it work as a link properly, we need to url encode the work item name in the console url properly.

**Comment by lukas-lansky** [comment]
@alexperovich You mean when you check the logs in the GUI of AzDO? (I'm somehow oblivious to that as I access them through the API more often than that.) True, escaping would be even better!

File: src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs [hunk 1]
```diff
@@ -37,8 +35,9 @@ protected override Task ExecuteCore(CancellationToken cancellationToken)
                         var jobName = workItem.GetMetadata("JobName");
                         var workItemName = workItem.GetMetadata("WorkItemName");
                         var consoleUri = workItem.GetMetadata("ConsoleOutputUri");
+                        var logUrl = System.Uri.EscapeUriString(consoleUri + accessTokenSuffix);
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs:38]: I'm surprised this works to properly escape the entire url. I thought we would have to escape the work item name where we construct this value. If this works then its good.

**Review by alexperovich — APPROVED** [review]

File: src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs (hunk 1 above)
Comment (lukas-lansky) [reply src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs:38]: I tried that for the URI in question and it translated into `https://helix.dot.net/api/2019-06-17/jobs/2db02a82-cc0b-4316-a9ee-e1489413e4a4/workitems/SDK%20ASP.NET%20MVC%20App%20Template%20netcoreapp3.1%20Clean%20Build/console`, which seems reasonable to me.

Comment (lukas-lansky) [reply src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs:38]: But you are probably right that it's better practice to escape during construction. Looking into [RFC 3986, section 2.4](view-source:https://tools.ietf.org/html/rfc3986#section-2.4), "Under normal circumstances, the only time when octets within a URI are percent-encoded is during the process of producing the URI from its component parts." And it makes sense: what if the work item name contains `/`?

Let's see if I am able to find a better fix tomorrow.

**Pull request closed without merging**